ui/src/screen/project/source.ts

```diff
@@ -394,20 +407,40 @@ export const fetchMergeRequestCommits = async (
 
   if (screen.status === remote.Status.Success) {
     const {
-      data: { project },
+      data: { peer, project },
     } = screen;
 
     try {
-      const revision: source.Sha = {
-        type: source.RevisionType.Sha,
-        sha: mergeRequest.commit,
-      };
-      const commits = await source.fetchCommits(
+      const baseProjectLatestCommit = (
+        await source.fetchCommits(project.urn, peer.peerId, {
+          type: source.RevisionType.Branch,
+          name: project.metadata.defaultBranch,
+        })
+      ).history[0];
+
+      const mrCommits = await source.fetchCommits(
         project.urn,
         mergeRequest.peer_id,
-        revision
+        {
+          type: source.RevisionType.Sha,
+          sha: mergeRequest.commit,
+        }
       );
-      mergeRequestCommitsStore.success(commits);
+
+      const nothingNewIdx = baseProjectLatestCommit
+        ? mrCommits.history.findIndex(
+            ch => ch.sha1 === baseProjectLatestCommit.sha1
+          )
+        : 0;
+      const newCommits = mrCommits.history.slice(
+        0,
+        nothingNewIdx === -1 ? 0 : nothingNewIdx
+      );
+
+      mergeRequestCommitsStore.success({
+        history: newCommits,
+        stats: { ...mrCommits.stats, commits: newCommits.length },
+      });
```

Comment:
I believe this `source.CommitHistory` object needs to be a `CommitHistory` object. Otherwise the app does not work. This also requires an adjustment of the type of the `mergeRequestCommits` store.

Reply:
Obsolete, result of a bad rebase, fixed now but please double check.